validator/main.go

```diff
@@ -614,24 +657,110 @@ func assertDriverContainerReady(silent bool) error {
 	return runCommand(command, args, silent)
 }
 
-func (d *Driver) runValidation(silent bool) (string, bool, error) {
-	driverRoot, isHostDriver := getDriverRoot()
-	if !isHostDriver {
-		log.Infof("Driver is not pre-installed on the host. Checking driver container status.")
-		if err := assertDriverContainerReady(silent); err != nil {
-			return "", false, fmt.Errorf("error checking driver container status: %v", err)
+// isDriverManagedByOperator determines if the NVIDIA driver is managed by the GPU Operator.
+// We check if at least one driver DaemonSet exists in the operator namespace that is
+// owned by the ClusterPolicy or NVIDIADriver controllers.
+func isDriverManagedByOperator(ctx context.Context) (bool, error) {
+	kubeConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return false, fmt.Errorf("error getting cluster config: %w", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return false, fmt.Errorf("error getting k8s client: %w", err)
+	}
+
+	opts := meta_v1.ListOptions{LabelSelector: labels.Set{appComponentLabelKey: "nvidia-driver"}.AsSelector().String()}
+	dsList, err := kubeClient.AppsV1().DaemonSets(namespaceFlag).List(ctx, opts)
+	if err != nil {
+		return false, fmt.Errorf("error listing daemonsets: %w", err)
+	}
+
+	for i := range dsList.Items {
+		ds := dsList.Items[i]
+		owner := meta_v1.GetControllerOf(&ds)
+		if owner == nil {
+			continue
+		}
+		if owner.Kind == nvidiav1.ClusterPolicyCRDName || owner.Kind == nvidiav1alpha1.NVIDIADriverCRDName {
```

Comment:
Makes this check more robust.
```suggestion
		if owner.APIVersion == "nvidia.com" && (owner.Kind == nvidiav1.ClusterPolicyCRDName || owner.Kind == nvidiav1alpha1.NVIDIADriverCRDName) {
```

Reply:
Wouldn't API version be `nvidia.com/v1` or `nvidia.com/v1alpha1`?

Reply:
What about?
```
strings.HasPrefix(owner.APIVersion, "nvidia.com/") && . . .
```

Reply:
That works!

Reply:
Done.